src/tree/updater_gpu_hist.cu

```diff
@@ -631,24 +473,23 @@ struct DeviceShard {
                           std::function<bool(ExpandEntry, ExpandEntry)>>;
   std::unique_ptr<ExpandQueue> qexpand;
 
-  DeviceShard(int _device_id, bst_uint _n_rows, TrainParam _param, uint32_t column_sampler_seed,
+  DeviceShard(int _device_id,
+              EllpackPageImpl* page,
+              bst_uint _n_rows,
+              TrainParam _param,
+              uint32_t column_sampler_seed,
               uint32_t n_features)
       : device_id(_device_id),
+        page(page),
```

Comment:
perhaps prefix the formal argument `page` with an underscore `_page`

Reply:
Done.